### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by mikem8361**
Fix some symbol publishing problems.

Fix issue #130 by increasing the section/program header count to 30000.

Fix issue #117. Always set the symbol file stream position to 0.

Add Equals, GetHashCode and ToString to the SymbolStore implementations

Fix issue #106.

### CODE REVIEW

File: src/Microsoft.FileFormats/ELF/ELFStructures.cs [hunk 1]
```diff
@@ -138,15 +138,15 @@ public ValidationRule IsProgramHeaderCountReasonable
...
-                return new ValidationRule("Unreasonably large ELFHeader ProgramHeaderCount", () => ProgramHeaderCount <= 10000);
+                return new ValidationRule("Unreasonably large ELFHeader ProgramHeaderCount", () => ProgramHeaderCount <= 30000);
             }
         }
 
         public ValidationRule IsSectionHeaderCountReasonable
         {
             get
             {
-                return new ValidationRule("Unreasonably large ELFHeader SectionHeaderCount", () => SectionHeaderCount <= 10000);
```
Comment (hoyosjs) [inline src/Microsoft.FileFormats/ELF/ELFStructures.cs]: What generates so many headers?

File: src/dotnet-symbol/README.md [hunk 2]
```diff
@@ -52,6 +52,8 @@ To verify a symbol package on a local VSTS symbol server:
 
 ## Notes ##
 
+Symbol download is only supported for official .NET Core runtime versions installed from our official [web site](https://dotnet.microsoft.com/download/dotnet-core). Runtimes from community sites like [archlinux](https://www.archlinux.org/packages/community/x86_64/dotnet-runtime/) are not support.
```
Comment (hoyosjs) [inline src/dotnet-symbol/README.md]:
```suggestion
Symbol download is only supported for official .NET Core runtime versions acquired through official channels such as [the official web site](https://dotnet.microsoft.com/download/dotnet-core) and the [default sources in the dotnet installation scripts](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-scripts). Runtimes obtained from community sites like [archlinux](https://www.archlinux.org/packages/community/x86_64/dotnet-runtime/) are not supported. 
```

File: src/Microsoft.FileFormats/ELF/ELFStructures.cs (hunk 1 above)
Comment (mikem8361) [reply src/Microsoft.FileFormats/ELF/ELFStructures.cs]: `createdump` generates them. `createdump` is also going to be changed not to generate all the
"not backed by memory" program header notes which will reduce the number of program headers significantly.

File: src/dotnet-symbol/README.md (hunk 2 above)
Comment (mikem8361) [reply src/dotnet-symbol/README.md]: Done.

**Pull request merged**